sdk/src/main/java/com/qonversion/android/sdk/internal/QRemoteConfigManager.kt

```diff
@@ -0,0 +1,69 @@
+package com.qonversion.android.sdk.internal
+
+import com.qonversion.android.sdk.dto.QRemoteConfig
+import com.qonversion.android.sdk.dto.QonversionError
+import com.qonversion.android.sdk.internal.services.QRemoteConfigService
+import com.qonversion.android.sdk.internal.services.QUserInfoService
+import com.qonversion.android.sdk.listeners.QonversionRemoteConfigCallback
+import javax.inject.Inject
+
+class QRemoteConfigManager @Inject internal constructor(
+    private val remoteConfigService: QRemoteConfigService,
+    private val userInfoService: QUserInfoService
+) {
+    private var currentRemoteConfig: QRemoteConfig? = null
+    private var isLaunchFinished: Boolean = false
+    private var remoteConfigCallbacks = mutableListOf<QonversionRemoteConfigCallback>()
+    private var isRequestInProgress: Boolean = false
+
+    fun launchFinished(finished: Boolean) {
+        isLaunchFinished = finished
+
+        if (finished && remoteConfigCallbacks.isNotEmpty() && !isRequestInProgress) {
+            loadRemoteConfig(null)
+        }
+    }
+
+    fun loadRemoteConfig(callback: QonversionRemoteConfigCallback?) {
+        currentRemoteConfig?.let {
+            callback?.onSuccess(it)
+            return
+        }
+
+        if (!isLaunchFinished || isRequestInProgress) {
+            callback?.let {
+                remoteConfigCallbacks.add(it)
+                return
+            }
+        }
+
+        isRequestInProgress = true
+        val userId = userInfoService.obtainUserID()
```

Comment:
I'm not sure this is a good way to get the user id. Let's instead inject `InternalConfig` in constructor and use `internalConfig.uid` here